[TA1304] unit tests for negative cases in target for network connections

mynktl opened the pull request:
Changes : 
- Added error injection support in `istgt_integration` test utility. 
    As of now, error injection is added for below scenarios : 
       1. Closing management or data connection from replica
       2. Sending multiple responses for a single management command from replica
       3. Sending errored response for IOs from replica

- Code fix for the errors reported by error injection feature.

Commit:
Test to verify negative scenarios for replica network interface

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

vishnuitta commented on src/replication.c:
```diff
@@ -1479,6 +1478,10 @@ accept_mgmt_conns(int epfd, int sfd)
 	int mgmt_fd;
 	mgmt_event_t *mevent1, *mevent2;
 
+/*
```
this is not required

mynktl replied:
ok

vishnuitta commented on src/replication.c:
```diff
@@ -2253,6 +2259,8 @@ handle_mgmt_conn_error(replica_t *r, int sfd, struct epoll_event *events, int ev
 				if (r_ev == r) {
 					TAILQ_REMOVE(&r->spec->rwaitq, r, r_waitnext);
 					r->conn_closed++;
+					if (r->io_resp_hdr)
```
set io_resp_hdr to NULL here, and also, in handle_data_conn_error() (with lock)

mynktl replied:
ok

Commit:
Addressing review comments

Signed-off-by: mayank <mayank.patel@cloudbyte.com>

mynktl commented:
Hi @vishnuitta
I have addressed all review comments given by you. I have also enabled error injection for IOs and added comments for replica structure variables.

vishnuitta reviewed (approved):
changes are good.
